examples/gno.land/r/sys/validators/gnosdk.gno

```diff
@@ -1,16 +1,24 @@
 package validators
 
-import "gno.land/p/sys/validators"
+import (
+	"sort"
 
-// getChanges returns the validator changes stored on the realm.
+	"gno.land/p/sys/validators"
+)
+
+// GetChanges returns the validator changes stored on the realm, since the given block number.
 // This function is unexported and intended to be called by gno.land through the GnoSDK
-func getChanges() []validators.Validator {
-	// Construct the changes
-	valsetChanges := make([]validators.Validator, len(changes))
-	copy(valsetChanges, changes)
+func GetChanges(from int64) []validators.Validator {
+	// Perform binary search to find the starting index
+	idx := sort.Search(len(changes), func(i int) bool {
```

Comment:
The risk of bricking contract can be mitigated by using an AVL tree with `seqid(std.GetHeight())` as the key.

Reply:
What risk of bricking do we have?

This AVL tree with have very few change entries throughout its lifetime -- the set is not going to be changing drastically on each block

Reply:
It has a potentially infinite size, as we continuously add new elements. Additionally, since we manage not only addition but also updates and deletions, the size ultimately depends on how the govDAO is active.

Another approach is to create a `GC()` function to call when the contract is bricked.

Reply:
I've switched to `avl` usage for the changes, and added appropriate unit tests:

[a16ca4b](https://github.com/gnolang/gno/pull/2229/commits/a16ca4b0548bdbc1cd64154eac0ccaec1abc19ab)